src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -21,6 +23,12 @@ export const TransactionContract = ({
   let assets = []
   if (txDetails.sellAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Source))
   if (txDetails.buyAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Destination))
+  const isReceive = txDetails.tradeTx?.type === TxType.Receive
+  const isWithdraw = txDetails.tx.data?.method === ContractMethods.Withdraw
+  const isSend = txDetails.tradeTx?.type === TxType.Send
+  const i18n = isReceive ? txDetails.tradeTx?.type : txDetails.tx.data?.method
+  const negated = [isWithdraw && isSend, false]
```

Comment:
This is a little strange - the second item of this array is always `false`, which is a hint that we should refactor.

We can change this to something like `firstAssetOutgoing`, which is of type `boolean`. Then in `TransactionGenericRow` we can check if we are mapping over the first asset, and if so, add a `-` to the `prefix`.

Reply:
This is less memory intensive! I had gone the least intrusive route.